alibi_detect/utils/missing_optional_dependency.py

```diff
@@ -25,7 +25,9 @@
     "tensorflow_probability": 'tensorflow',
     "tensorflow": 'tensorflow',
     "torch": 'torch',
-    "pytorch": 'torch'
+    "pytorch": 'torch',
+    "keops": 'keops',
+    "pykeops": 'keops',
```

Comment:
I don't actually understand why we need both `"keops": 'keops',` and `"pykeops": 'keops',` (or `"torch": 'torch',` and `    "pytorch": 'torch',` for that matter). Is this necessary? Perhaps a docstring or a comment for `ERROR_TYPES` is needed to clarify for the future (separate PR). @mauicv

Reply:
Yeah, good idea.  

Basically in the code we import from `pykeops` whereas when setting the backend the user specifies `keops` so somewhere we need to map between these two and it's here in the `ERROR_TYPES` dict. If the entry above is `keops` then the `impot_optional` will fail to catch the error when it imports from `pykeops`. If we change the above to `pykeops` then `import_optional` will catch the error but the Backend Validator will get a key error as `keops` is what the user specifies. So we could try and set a convention to only use one of `pykeops` or `keops` everywhere but I felt it would be more robust to just use a surjective mapping `ERROR_TYPES`.  

I'll make an issue and create another PR fixng this and some other stuff!